modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/AppsFlyerAnalyticsWrapper.kt

```diff
@@ -0,0 +1,67 @@
+package au.com.shiftyjelly.pocketcasts.analytics
+
+import android.content.Context
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import com.appsflyer.AppsFlyerLib
+import com.appsflyer.attribution.AppsFlyerRequestListener
+import java.util.concurrent.atomic.AtomicBoolean
+import timber.log.Timber
+
+open class AppsFlyerAnalyticsWrapper(
+    private val appsFlyerLib: AppsFlyerLib?,
+    private val context: Context,
+) {
+    private val started = AtomicBoolean(false)
+    private val logEventListener = object : AppsFlyerRequestListener {
+        override fun onSuccess() {
+            Timber.i("AppsFlyer event logged successfully")
+        }
+
+        override fun onError(code: Int, descrption: String) {
+            LogBuffer.e("AppsFlyer", "AppsFlyer event logging failed. Error code: $code, description: $descrption")
+        }
+    }
+    private val launchEventListener = object : AppsFlyerRequestListener {
+        override fun onSuccess() {
+            Timber.i("AppsFlyer launch successful")
+        }
+
+        override fun onError(code: Int, descrption: String) {
+            LogBuffer.e("AppsFlyer", "AppsFlyer launch failed. Error code: $code, description: $descrption")
+        }
+    }
+
+    open fun logEvent(name: String, params: Map<String, Any>, userId: String) {
+        if (appsFlyerLib == null) {
+            return
+        }
+        if (!started.get()) {
+            startAppsFlyer(context, userId)
+        }
+        appsFlyerLib.logEvent(context, name, params, logEventListener)
+    }
+
+    open fun startAppsFlyer(context: Context, userId: String) {
```

Comment:
Should we use the context that it's already available in the class property instead of passing it?

Reply:
Thanks.